messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/deployment/injection/InjectedJMSContext.java

```diff
@@ -86,6 +87,18 @@
     JMSContext getDelegate() {
         boolean inTx = isInTransaction();
         AbstractJMSContext jmsContext = inTx ? transactedJMSContext.get() : requestedJMSContext;
+        //fix of  WFLY-9501
+        // CCM tries to clean opened connections before execution of @PreDestroy method on JMSContext - which is executed after completion, see .
+        // Correct phase to call close is afterCompletion {@see TransactionSynchronizationRegistry.registerInterposedSynchronization}
+        if(inTx) {
+            //to avoid registration of more listeners for one context, flag in transaction is used.
+            Object alreadyRegistered = transactionSynchronizationRegistry.getResource(AfterCompletionSynchronization.class.getName());
```

Comment:
Why don't you use the jmsContext as the resource?

Reply:
JmsContext as the resource could be used without a problem. I've just used boolean as a typical type for flags.
But because I'm only testing existence of this flag, boolean is not needed, I'll use jmsContext.
Thank you for warning me about this.